org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/ZigBeeConverterGenericButton.java

```diff
@@ -74,41 +75,65 @@ public synchronized boolean initializeConverter() {
         for (CommandSpec commandSpec : handledCommands.values()) {
             int clusterId = commandSpec.getClusterId();
 
-            if (clientClusters.stream().anyMatch(cluster -> cluster.getClusterId().intValue() == clusterId)) {
-                // bind to each output cluster only once
-                continue;
-            }
+            boolean success = commandSpec.isClientCluster() ?
+                    bindCluster(
+                            "client", clientClusters, clusterId,
+                            endpoint::getOutputCluster, cluster -> cluster.addCommandListener(this))
+                    :
+                    bindCluster(
+                            "server", serverClusters, clusterId,
+                            endpoint::getInputCluster, cluster -> cluster.addAttributeListener(this));
 
-            ZclCluster clientCluster = endpoint.getOutputCluster(clusterId);
-            if (clientCluster == null) {
-                logger.error("{}: Error opening client cluster {} on endpoint {}", endpoint.getIeeeAddress(), clusterId,
-                        endpoint.getEndpointId());
+            if (!success) {
                 return false;
             }
+        }
 
-            try {
-                CommandResult bindResponse = bind(clientCluster).get();
-                if (!bindResponse.isSuccess()) {
-                    logger.error("{}: Error 0x{} setting client binding for cluster {}", endpoint.getIeeeAddress(),
-                            toHexString(bindResponse.getStatusCode()), clusterId);
-                }
-            } catch (InterruptedException | ExecutionException e) {
-                logger.error(endpoint.getIeeeAddress() + ": Exception setting binding to cluster " + clusterId, e);
-            }
+        return true;
+    }
 
-            clientCluster.addCommandListener(this);
-            clientClusters.add(clientCluster);
+    private boolean bindCluster(String clusterType, Collection<ZclCluster> existingClusters, int clusterId,
+                                Function<Integer, ZclCluster> clusterFunction,
```

Comment:
small suggestion: rename `clusterFunction` to `getClusterByIdFunction` (or some other name telling what the function does). WDYT?